app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventMapFragment.java

```diff
@@ -1,18 +1,146 @@
+import ch.epfl.sweng.eventmanager.ui.eventSelector.EventPickingActivity;
+import ch.epfl.sweng.eventmanager.ui.eventShowcase.models.EventShowcaseModel;
 
 /**
  * A simple {@link Fragment} subclass.
  * Activities that contain this fragment must implement the
  * to handle interaction events.
  */
-public class EventMapFragment extends AbstractShowcaseFragment {
+public class EventMapFragment extends Fragment {
     private static final String TAG = "EventMapFragment";
+    private EventShowcaseModel model;
+    private GoogleMap mMap;
+    private ClusterManager<Spot> mClusterManager;
 
     public EventMapFragment() {
         // Required empty public constructor
-        super(R.layout.fragment_event_map);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        model = ViewModelProviders.of(getActivity()).get(EventShowcaseModel.class);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_event_map, container, false);
+
+
+
+        SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager()
+                .findFragmentById(R.id.mapFragment);
+        if (mapFragment == null) {
+            FragmentManager fragmentManager = getFragmentManager();
+            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+            mapFragment = SupportMapFragment.newInstance();
+            fragmentTransaction.replace(R.id.mapFragment, mapFragment).commit();
+        }
+        if (mapFragment != null) {
+            mapFragment.getMapAsync(new OnMapReadyCallback() {
+                @Override
+                public void onMapReady(GoogleMap googleMap) {
+                    mMap = googleMap;
+                    if (mMap != null) {
+                        float zoomLevel = 19.0f; //This goes up to 21
+                        model.getLocation().observe(getActivity(), loc -> {
+                            if (loc != null) {
+                                LatLng place = new LatLng(loc.getLatitude(), loc.getLongitude());
+                                mMap.addMarker(new MarkerOptions().position(place).title(loc.getName()));
+                                mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(place, zoomLevel));
+
+                                setUpClusterer(setFakeSpot());
+                                mMap.getUiSettings().setCompassEnabled(true);
+                                mMap.getUiSettings().setZoomControlsEnabled(true);
+                                mMap.getUiSettings().setMapToolbarEnabled(true);
+                            }
+                        });
+                    }
+
+                }
+            });
+
+        }
+
+        return view;
+    }
+
+    private List<Spot> setFakeSpot() {
```

Comment:
Not sure if this should remain here

Reply:
yes, we have to do it for the next print